### PR TITLE
854 arpspoof missing components

### PR DESCRIPTION
**Pull request by vtruglio**
As per the attached issue, this request aimed to implement previously missing components into the ARPSpoof tools script, as well as correct misplaced dependencies referencing.

Concerning the pkexec delay issue, this was responded to with recommendations in coinciding documentation for submission.

### CODE REVIEW

**Review by JacksonGIThuba — COMMENTED** [review]
Did manage to get it to work but the process isn't efficient enough to be practical. I changed the number of CPUs on my vm to 5 and it still ran extremely slow becoming quite laggy. It did cause my OS to crash after installing the dependencies.

**Comment by vtruglio** [comment]
Thanks for the quick reply Jackson.

Unfortunately, the Pkexec issue isn't something that can't be fixed with code implementation at the moment. The inefficiency of the process appears to be a fault with the machine that causes a huge delay in the DDT. As such, I added the required missing components but for the secondary issue of the actual process, I instead made some recommendations as to what could be adopted to fix this issue in the future in my OnTrack submission.

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 1]
```diff
@@ -49,7 +50,7 @@ const ARPSpoofing = () => {
         "Step 4: View the output block below to view the results.";
 
     // Component Constants.
-    const dependencies = ["dsniff"]; // Contains the dependencies required for the component.
+    const dependencies = ["arpspoof"]; // Contains the dependencies required for the component.
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:53]: Sorry @vtruglio. this is my bad. I was confused about what the dependencies variable should store, but I have subsequently tested it and can confirm that it should be the name of the package that requires installation. Therefore, please change back to dsniff. 
```suggestion
    const dependencies = ["dsniff"]; // Contains the dependencies required for the component.
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @vtruglio 

Sorry about the misdirection about the dependencies variable. I gave you a bum steer on that one! 

You will need to add the `<stack>` and `LoadingOverlayAndCancelButtonPkexec` components to the UI. See AirbaseNG for the following example:

image

Kind regards 

Michael

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 2]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
-                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
-                    <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
+                    <Stack>
+                        {LoadingOverlayAndCancelButtonPkexec(
+                            loading,
+                            pidGateway,
+                            handleProcessData,
+                            handleProcessTermination
+                        )}
+                        {LoadingOverlayAndCancelButtonPkexec(
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:200]: You have included the loading overlay twice. Need to remove the 2nd one. 
```suggestion
                    
```

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 3]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
+                            loading,
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:201]:
```suggestion
                         
```

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 4]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
+                            pidTarget,
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:202]:
```suggestion
                         
```

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 5]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
+                            handleProcessData,
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:203]:
```suggestion
                          
```

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 6]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
+                            handleProcessTermination
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:204]:
```suggestion
                          
```

File: src/components/ArpSpoof/ArpSpoof.tsx [hunk 7]
```diff
@@ -190,11 +190,25 @@ const ARPSpoofing = () => {
...
+                        )}
```
Comment (michaeljpigott) [inline src/components/ArpSpoof/ArpSpoof.tsx:205]:
```suggestion
                      
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @vtruglio 

This is very close. You just accidentally added the loading overlay twice. Can you please remove the 2nd one? 

Kind regards

Michael

**Comment by vtruglio** [comment]
Thanks Michael.

I've added two loading overlays as the tool utilises both a pidGateway and pidTarget variables. When testing with these both overlays, the tool did perform arpspoofing and return an output.

**Comment by michaeljpigott** [comment]
Hi @vtruglio, 

Good point - I forgot about that! 

Michael

**Review by michaeljpigott — APPROVED** [review]
Hi @vtruglio 

Thank you for making those changes. The tool looks a lot better now. 

Kind regards

Michael

**Pull request merged**